### PR TITLE
Fix max attribute on qty input of an out of stock attribute available on create order

### LINKED ISSUE #20670: BO - Orders page - increment/decrement button doesn't work when we try to add a product out of stock (allow order)
#### Describe the bug

In the BO => Orders page => New Order page => when we try to add an out of stock product with allowing order, the increment/decrement button doesn't work

#### Steps to Reproduce

1. Go to BO => Catalog => Products page
2. Add a product with quantity <= 0 with allow order selected
3. Go to BO => Orders => New Order page
4. Add the last product created
5. check the increment/decrement button
6. see error => we can't increment the product quantity

**But if you try to enter directly the quantity in the field and click on "Add to cart", it's working**

**Screenshots**

image
https://drive.google.com/file/d/10ob-O_t4_JSv1hhM5HweRPZ4J5IMcuuE/view?usp=sharing

#### Additional information

* PrestaShop version: 1770beta2, 1.7.7.x
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       |  1.7.7.x
| Description?  | You couldnt use arrows when creating an order of a product which has 0 stocks but is available for order
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #20670 .
| How to test?  | See issue

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20714)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/pages/order/create/product-renderer.js [hunk 1]
```diff
@@ -168,9 +170,14 @@ export default class ProductRenderer {
...
-  renderStock(stock) {
+  renderStock(stock, infinitMax) {
     $(createOrderMap.inStockCounter).text(stock);
-    $(createOrderMap.quantityInput).attr('max', stock);
+
+    if (!infinitMax) {
+      $(createOrderMap.quantityInput).attr('max', stock);
+    } else {
+      $(createOrderMap.quantityInput).removeAttr('max');
+    }
```
Comment (NeOMakinG) [inline admin-dev/themes/new-theme/js/pages/order/create/product-renderer.js:180]: Here are main modifications

File: admin-dev/themes/new-theme/js/pages/order/create/product-renderer.js [hunk 2]
```diff
@@ -158,7 +160,7 @@ export default class ProductRenderer {
    * @param product
    */
   renderProductMetadata(product) {
-    this.renderStock(product.stock);
+    this.renderStock(product.stock, product.stock === 0 && product.availableOutOfStock ? true : false);
```
Comment (NeOMakinG) [inline admin-dev/themes/new-theme/js/pages/order/create/product-renderer.js:163]: One modification (others are linting)

**Review by atomiix — DISMISSED** [review]

File: admin-dev/themes/new-theme/js/pages/order/create/product-renderer.js (hunk 2 above)
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/pages/order/create/product-renderer.js:163]:
```suggestion
    this.renderStock(product.stock, product.stock === 0 && product.availableOutOfStock);
```

**Review by PierreRambaud — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Comment by ngodefroy** [comment]
Hello @NeOMakinG @marionf 
I was able to reproduce the following but not sure if ordering a product with no stock is to be permitted?
On my case, I was able to order a product by incrementing it or manually set the desired quantity.

_Note: If I try to add a second time the product, I get an error that tells me the product is out of stock._

https://drive.google.com/file/d/1gUMpclyiyUBWGbIIDONJC3caRyCsvKgd/view

Thank you

**Comment by NeOMakinG** [comment]
It's permitted only if you check something on the product (allow order when out of stock)

**Comment by ngodefroy** [comment]
> It's permitted only if you check something on the product (allow order when out of stock)

Ok, so I shouldn't get the error when adding products two times, I guess..

**Comment by NeOMakinG** [comment]
Well, but I doubt this is related to the same issue :/

**Comment by ngodefroy** [comment]
> Well, but I doubt this is related to the same issue :/

Ok ! I'm validating this one and opening a new issue. Thank you

**Review by matks — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @NeOMakinG & @ngodefroy

**Pull request merged**